File: 프로그래머스/src/카펫.cpp
#include <string>
#include <vector>
#include <iostream>
using namespace std;
// https://school.programmers.co.kr/learn/courses/30/lessons/42842
vector<int> brute_force(int size, int brown) {
    vector<int> answer = {0,0};
    for (int i = 1; i <size/2 ; i++) {
        int width = i;
        int height = size/i;
        if(size%i==0) {
            if(width>height) break;
            // 테두리 한줄은 갈색임 -> (width + height) *2 - 4 = brown 이여야함
            if(((width+height)*2-4) == brown) {
                answer[0] = height;
                answer[1] = width;
            }
        } 
    }
    return answer;
}
vector<int> solution(int brown, int yellow) {
    return brute_force(brown+yellow,brown);
}

File: 2026/완전탐색/카펫.py
# https://school.programmers.co.kr/learn/courses/30/lessons/42842?language=python3

# brown : 가로 x 2 + 세로 x 
# 10 = (4 x 2) + (3 x 2)  - 4  = 8 + 6 - 4
# 8 = (3 x 2) + (3 x 2) - 4  = 6 + 6 - 4 
# 24 = (8 x 2) + (6 x 2) - 4 = 16 + 12 - 4 = 24

# brown = (width x 2) + (height x 2) - 4
# brown + 4 = width x 2 + height x 2 + 2(w+h)
# also, brown is always even with my intuition
# (brown+4)/2 = w+h

# 2nd observation : width x height = sum(brown,yellow)

def solution(brown, yellow):
    answer = []
    sum_wh = int((brown+4)/2)
    area = brown + yellow
    print(sum_wh)
    
    # 가로 >= 세로
    # observation3 
    # BROWN = 8 이상임. 즉, height 는 3 이상임.
    for height in range(3, sum_wh+1):
        width = sum_wh - height
        if width * height == area:
            answer = [width, height]
            break;
    return answer

